src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,127 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+import java.util.Objects;
+
 public class ArrayList<T> implements List<T> {
+    private static final int STANDARD_CAPACITY = 10;
+    private T[] arrayList;
+    private int size;
+
+    public ArrayList() {
+        arrayList = (T[]) new Object[STANDARD_CAPACITY];
+    }
+
     @Override
     public void add(T value) {
-
+        if (hasArraySpace()) {
+            arrayList[size] = value;
+            size++;
+            return;
+        }
+        extendCapacity();
+        arrayList[size] = value;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        if (!hasArraySpace()) {
+            extendCapacity();
+        }
+        if (index < 0 || index > arrayList.length || index > size + 1) {
+            throw new ArrayListIndexOutOfBoundsException("Your index does not suit rules");
+        }
+        System.arraycopy(arrayList, index, arrayList, index + 1, size - index);
+        arrayList[index] = value;
+        size++;
 
     }
 
     @Override
     public void addAll(List<T> list) {
-
+        if (list != null) {
+            T[] addArray = (T[]) new Object[list.size()];
+            for (int i = 0; i < list.size(); i++) {
+                addArray[i] = list.get(i);
+            }
+            while (arrayList.length < size + addArray.length) {
+                extendCapacity();
+            }
+            System.arraycopy(addArray, 0, arrayList, size, addArray.length);
+            size = size + addArray.length;
```

Comment:
this can be simplified up to several lines. HINT: you have `add(T value)` method

Reply:
add by value has been used